src/EditorFeatures/CSharpTest/UsePatternMatching/CSharpAsAndNullCheckTests.cs

```diff
@@ -874,6 +874,403 @@ string M(object o, bool b)
+@"class C
+{
+    void M(object e, C x)
+    {
+        [|var|] c = e as C;
+        for (;(c)!=null;) {}
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestForStatement_InlineTypeCheck()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e)
+    {
+        [|C|] c;
+        for (; !((c = e as C)==null);) { }
+    }
+}",
+@"class C
+{
+    void M(object e)
+    {
+        for (; !(!(e is C c));) { }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestForStatement_InScope()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e)
+    {
+        [|C|] c = null;
+        for (; !((c = e as C)==null);)
+        {
+            M(c);
+        }
+    }
+}",
+@"class C
+{
+    void M(object e)
+    {
+        for (; !(!(e is C c));)
+        {
+            M(c);
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestForStatement_NotAssignedBeforeAccess()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e)
+    {
+        [|C|] c = null;
+        for (; ((c = e as C)==null);)
+        {
+            if (b) c = null;
+            M(c);
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestForStatement_AssignedBeforeAccess()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e, bool b)
+    {
+        [|C|] c = null;
+        for (; (c = e as C)==null;)
+        {
+            if (b) c = null;
+            else c = null;
+            M(c);
+        }
+    }
+}",
+@"class C
+{
+    void M(object e, bool b)
+    {
+        for (; !(e is C c);)
+        {
+            if (b) c = null;
+            else c = null;
+            M(c);
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestForStatement_MultipleDeclarators()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e)
+    {
+        [|C|] c = null, x = null;
+        for (; !((c = e as C)==null);)
+        {
+            M(c);
+        }
+    }
+}",
+@"class C
+{
+    void M(object e)
+    {
+        C x = null;
+        for (; !(!(e is C c));)
+        {
+            M(c);
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestForStatemnet_UseBeforeDeclaration()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(object e)
+    {
+        [|C|] c = null, x = c;
+        for (; !((c = e as C)==null);)
```

Comment:
was a for-block really needed for all these tests?

Reply:
That was new so I based new tests on for